Update Chord Diagram example to make more sense

jens-ox opened the pull request:
This PR updates the chord diagram.

1. The linear gradients are replaced by static colors to better show what is going on.
2. The coloring of the ribbons is fixed. Previously, the coloring function was used identically for the arcs and ribbons (using the index), but that doesn't work for ribbons, as the index of a ribbon is not related to the index of the arcs (resulting in a ribbon coloring that doesn't make sense). Now, the index of the target group is used to color the ribbons.
3. The code example of the chord graph is updated accordingly.

**Old**:
Notice how the coloring of the ribbons with identical source and target group is different than the coloring of that group. I suppose that this wasn't noticed because of the gradients :smile:
![Selection_003](https://user-images.githubusercontent.com/5368029/71678453-7c727e80-2d85-11ea-8aaa-54242cb64477.png)

**New**:
![Selection_004](https://user-images.githubusercontent.com/5368029/71678451-7bd9e800-2d85-11ea-98b3-35200011c0b0.png)

jens-ox commented:
The color scale is dark2 from d3-scale-chromatic which is already used in vx-demo.

williaster commented:
Hey @jens-ox, thanks for checking out `vx` and making this demo more interpretable! 🙏 I agree that the color should deterministically come from the target (or source), not the arc index. 

My only concern is over the new color palette. We try to make the gallery quite vibrant and visually intriguing, and the dark d3 color palette feels less engaging to me. We also often create our own color palettes simply to promote diversity in color beyond what d3 provides, as d3 colors are used everywhere and therefore feel less unique. 

Would you be open to leveraging the previous colors? I would also love to keep the gradients because they are more visually interesting, I think we could make them interpretable by toning back the variation within each gradient (i.e., make the colors within a single gradient closer to each other, like same hue with varied brightness). I pasted some possible variations below but feel free to tweak them!

![image](https://user-images.githubusercontent.com/4496521/71689950-26d3ad00-2d58-11ea-92ee-05a6f7a74e8c.png)

williaster commented:
also not sure what's up with the coverage check, don't worry about that.

jens-ox commented:
Thanks for getting back! I agree, the gradients were much more vibrant than dark2.

I tried the colors you proposed. As there are only 4 groups in the chart, I had to leave one color out:

Without orange           |  Without blue
:-------------------------:|:-------------------------:
![Selection_007](https://user-images.githubusercontent.com/5368029/71698940-aba4e200-2dbd-11ea-88fd-b874b545f5d4.png)  |  ![Selection_006](https://user-images.githubusercontent.com/5368029/71698941-ac3d7880-2dbd-11ea-8adc-97ede75a3170.png)

In my opinion, this might be a little bit too colorful. I think the "old" gradients look better:
![Selection_005](https://user-images.githubusercontent.com/5368029/71699009-f45c9b00-2dbd-11ea-9859-fe06315f6e48.png)

I switched the PR back to the old gradients. Also, the domain of the color scale was erroneous, fixed that.

williaster commented on packages/vx-demo/src/pages/chord.js:
```diff
@@ -19,23 +19,15 @@ import { Arc } from '@vx/shape';
 import { Chord, Ribbon } from '@vx/chord';
 import { scaleOrdinal } from '@vx/scale';
-import { LinearGradient } from '@vx/gradient';
+import { schemeDark2 } from 'd3-scale-chromatic';
```
could we remove the changes in the page source code as well? 

sorry this is more work than it should be, we're scheming ways to improve docs + examples in the near future after TS wraps up.

jens-ox replied:
Sorry, forgot about that one. Updated the page source code.

williaster commented:
Thanks @jens-ox! Heads up that Harry is out for a few weeks and I don't have permissions to publish the gallery site, so this will likely not be visible immediately.
